atomic_reactor/plugins/post_koji_upload.py

```diff
@@ -324,8 +327,17 @@ def get_digests(self):
                     else:
                         digest_list = [self.select_digest(image_digests)]
                     digests[image.to_str(registry=False)] = digest_list
-
-        return digests
+                    digest_images[image.to_str(registry=False)] = image_digests
+                if image_str in digest_images:
+                    digest = digest_images[image_str]
+                    for digest_version in digest.content_type:
+                        if digest_version not in digest:
+                            continue
+                        digest_type = get_manifest_media_type(digest_version)
+                        docker_digests[digest_type] = digest[digest_version]
+
+        self.log.debug("docker_digests is -%s-", docker_digests)
+        return digests, docker_digests
```

Comment:
It would feel a little cleaner if this could return a dict mapping image names (`image.to_str(registry=False)`) to a tuple of (media_type, digest). Then get_repositories() could be fed a result based on that, e.g. `{key: [digest for _, digest in typed_digests] for key, typed_digests in digests.items()}`.

Reply:
I honestly don't see how that's cleaner. Instead of returning the two dictionaries that we need, with different keys, we return one dictionary that needs to be parsed for each use case.

my code is extra.docker.repositories = get_repositories(digests)
extra.docker.digests = docker_digests

your code is extra.docker.repositories = get_repositories({key: [digest for _, digest in typed_digests] for key, typed_digests in digests.items()})
extra.docker.digests = {type_digests[1]: key for key, typed_digests in digests.items()}

I may have gotten the parsing for extra.docker.digests wrong there.

Reply:
Something like that, yes, although I'd write the second one something like `{media_type: digest for media_type, digest in digests.values()}`.

The thing that I'm trying to make clearer/cleaner is "return digests, docker_digests", which doesn't make it very clear what the difference is between the two items in that returned tuple.

To my mind, the dict comprehensions make it much clearer what the metadata represents. Just me?

Reply:
Oh, the variable names are completely unclear. I agree that's a problem that I have often been struggling with during my experience with this project.
The digests return of get_digests is only ever fed to get_repositories. Would it help to merge get_digests and get_repositories into one function, get_docker_repositories_and_digests, that returns the dict that will be assigned to extras.docker.repositories and the dict that will be assigned to extra.docker.digests? That seems cleaner to me.